sdk/go/pulumi/provider.go

```diff
@@ -202,21 +224,316 @@ func constructInputsMap(ctx *Context, inputs map[string]interface{}) (Map, error
 		known := !ci.value.ContainsUnknowns()
 		value, secret, err := unmarshalPropertyValue(ctx, ci.value)
 		if err != nil {
-			return nil, errors.Wrapf(err, "unmarshaling input %s", k)
+			return nil, fmt.Errorf("unmarshaling input %q: %w", k, err)
 		}
 
 		resultType := anyOutputType
 		if ot, ok := concreteTypeToOutputType.Load(reflect.TypeOf(value)); ok {
 			resultType = ot.(reflect.Type)
 		}
 
-		output := ctx.newOutput(resultType, ci.deps...)
+		output := ctx.newOutput(resultType, ci.Dependencies(ctx)...)
 		output.getState().resolve(value, known, secret, nil)
 		result[k] = output
 	}
 	return result, nil
 }
 
+func gatherDeps(v resource.PropertyValue, deps urnSet) {
```

Comment:
The switch is complete? This is more my Go competency question I guess. I find this:

```
// PropertyValue is the value of a property, limited to a select few types (see below).
type PropertyValue struct {
	V interface{}
}
```

I'm trying to "see below" but being left a bit fuzzy with the exact set of types we allow here. In e.g. F# the type def itself would be a tight closed grammar (I think TypeScript can do it to).

For example I don't see Asset/Archive here but that's probably ok if they don't carry deps.

Reply:
I think so. I'm basing this off of two things:

1. [`PropertyValue.ContainsUnknowns`](https://github.com/pulumi/pulumi/blob/4f3f3666952b016c017e546ba4f40270246fa029/sdk/go/common/resource/properties.go#L377) and [`PropertyValue.ContainsSecrets`](https://github.com/pulumi/pulumi/blob/4f3f3666952b016c017e546ba4f40270246fa029/sdk/go/common/resource/properties.go#L395) for drilling into `PropertyValue`s:

2. How we gather dependencies in SDKs during marshaling. For example, [in Go](https://github.com/pulumi/pulumi/blob/4f3f3666952b016c017e546ba4f40270246fa029/sdk/go/pulumi/rpc.go#L231) and the other SDKs, we only add to the dependencies when marshaling `Output`s and resource references (adding the resource's URN as a dependency).